docs/ORTModule_ModuleWithLoss_Wrapper.md

```diff
@@ -0,0 +1,114 @@
+# ONNX Runtime ModuleWithLoss Wrapper
+
+This document provides instructions on implementing a wrapper similar to the ModuleWithLoss Wrapper in Optimum. By implementing this wrapper, you can compute the loss inside ONNX Runtime (ORT), enabling you to leverage additional optimizations such as label sparsity optimization.
+
+#### ModuleWithLoss Wrapper
+
+- **Description**: The ModuleWithLoss wrapper is a class that extends the nn.Module class from PyTorch. It is designed to enhance the functionality of an existing model by allowing the computation of loss inside the wrapper, leveraging the benefits of ONNX Runtime (ORT) and enabling additional optimizations such as label sparsity optimization.
+
+These are the Major changes on the code, Refer to these snippets as a reference while writing your own implementation.
+
+```python
+class ModuleWithLoss(nn.Module):
+    def __init__(self, model, args, label_smoother):
+        super().__init__()
+        self._original_model = model
+        self.args = args
+        # Label smoothing
+        self.label_smoother = label_smoother
+
+    def forward(self, inputs: Dict[str, Union[torch.Tensor, Any]], return_outputs):
+        # The compute_model_plus_loss_internal is assigned once the class is instantiated.
+        # It should have same signature as Trainer.compute_loss().
+        # We do this to avoid potential un-synced states if we duplicated compute loss codes .
+        return self.compute_model_plus_loss_internal(self._original_model, inputs, return_outputs)
+
+    @property
+    def module(self):
+        """The original `torch.nn.Module` that this module wraps.
+        This property provides access to methods and properties on the original module."""
+
+        return self._original_model.module
+
+    @property
+    def config(self):
+        return self._original_model.config
+
+def create_model_with_loss(self):
+    model_with_loss = ModuleWithLoss(self.model, self.args, self.label_smoother)
+    model_with_loss.compute_model_plus_loss_internal = types.MethodType(Trainer.compute_loss, model_with_loss)
+
+    return model_with_loss
+
+def _set_signature_columns_if_needed(self):
```

Comment:
Can we instead override `_set_signature_columns_if_needed(...)` inside the ModuleWithLoss base class?

Reply:
same as previous